File: src/main/java/shopping/wish/domain/WishEntity.java
package shopping.wish.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import shopping.common.domain.BaseTimeEntity;

@Getter
@NoArgsConstructor(access = lombok.AccessLevel.PROTECTED)
@Entity
@Table(name = "wishes")
public class WishEntity extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "member_id", nullable = false)
    private Long memberId;

    @Column(name = "product_id", nullable = false)
    private Long productId;

    public WishEntity(Long memberId, Long productId) {
        this.memberId = memberId;
        this.productId = productId;
    }
}

File: src/main/kotlin/shopping/storage/db/WishEntity.kt
package shopping.storage.db

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import shopping.domain.Wish

@Entity
@Table(name = "wishes")
class WishEntity(
    // NOTE: @ManyToOne 연관 매핑 대신 ID만 보관하여 도메인 간 결합도를 낮춥니다.
    //       멀티 모듈 분리 시 모듈 간 Entity 직접 참조를 방지할 수 있습니다.
    @Column(nullable = false)
    val memberId: Long,
    @Column(nullable = false)
    val productId: Long,
) : BaseEntity() {
    fun toDomain(): Wish =
        Wish(
            id = this.id,
            memberId = this.memberId,
            productId = this.productId,
        )

    companion object {
        fun from(wish: Wish): WishEntity =
            WishEntity(
                memberId = wish.memberId,
                productId = wish.productId,
            )
    }
}
